cli/src/semgrep/dependency_aware_rule.py

```diff
@@ -122,8 +123,8 @@ def generate_unreachable_sca_findings(
                         )
                     },
                 )
-                non_reachable_matches.append(match)
-    return non_reachable_matches, dep_rule_errors
+                non_reachable_matches.add(match)
+    return sorted(non_reachable_matches), dep_rule_errors
```

Comment:
Is the idea here to convert `RuleMatchSet` into a `List[RuleMatch]` to fit the function's return type?

Reply:
yep